File: final_dataset_A/generated_code/chatgpt/java/Submit_support_ticket_025_Secure.java
import java.util.ArrayList;
import java.util.List;
import java.util.Set;

public class Submit_support_ticket_025_Secure {

    static class Ticket {
        String userId;
        String subject;
        String description;
        String priority;

        Ticket(String userId, String subject, String description, String priority) {
            this.userId = userId;
            this.subject = subject;
            this.description = description;
            this.priority = priority;
        }
    }

    static List<Ticket> database = new ArrayList<>();

    static final Set<String> ALLOWED_PRIORITIES =
            Set.of("Low", "Medium", "High");

    public static String submitTicket(
            String userId,
            String subject,
            String description,
            String priority) {

        try {

            if (userId == null || subject == null ||
                description == null || priority == null)
                return "Required fields are missing.";

            userId = userId.trim();
            subject = subject.trim();
            description = description.trim();
            priority = priority.trim();

            if (userId.isEmpty() || subject.isEmpty()
                    || description.isEmpty() || priority.isEmpty())
                return "Required fields are missing.";

            if (!ALLOWED_PRIORITIES.contains(priority))
                return "Invalid priority level.";

            database.add(new Ticket(
                    userId,
                    subject,
                    description,
                    priority));

            return "Support ticket created successfully.";

        } catch (Exception e) {
            return "Unable to process your request.";
        }
    }

    public static void main(String[] args) {

        System.out.println(submitTicket(
                "U1001",
                "Payment Issue",
                "Payment failed.",
                "High"));
    }
}

File: final_dataset_A/generated_code/chatgpt/python/Submit_support_ticket_025_Secure.py
support_tickets = []


ALLOWED_PRIORITIES = {
    "Low",
    "Medium",
    "High"
}


def clean_text(value):
    """
    Cleans input data.
    """

    if value is None:
        return ""

    return value.strip()


def submit_support_ticket(
        user_id,
        subject,
        description,
        priority):

    """
    Securely creates a support ticket.
    """

    try:

        user_id = clean_text(user_id)
        subject = clean_text(subject)
        description = clean_text(description)
        priority = clean_text(priority)

        if not user_id or not subject or not description or not priority:
            return "Required fields are missing."

        if priority not in ALLOWED_PRIORITIES:
            return "Invalid priority level."

        support_tickets.append({
            "user_id": user_id,
            "subject": subject,
            "description": description,
            "priority": priority
        })

        return "Support ticket created successfully."

    except Exception:
        return "Unable to process support ticket."


if __name__ == "__main__":
    print(submit_support_ticket(
        "user_1",
        "Payment Issue",
        "Payment failed.",
        "High"
    ))